core/src/scene/light.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+ 
+#include <memory>
+#include "glm/glm.hpp"
+#include "util/shaderProgram.h"
+
+typedef enum {
+    DEFAULT,
+    VERTEX,
+    FRAGMENT,
+    BOTH
```

Comment:
I'm in favor of removing the 'BOTH' injection mode. The result of this mode is equivalent to adding two identical lights to the scene, one in 'VERTEX' mode and one in 'FRAGMENT' mode. The resulting 'light' will not behave as expected and I don't think there's a good use case for this.